packages/plugin-interactive-tools/sources/commands/search.tsx

```diff
@@ -0,0 +1,239 @@
+import {renderForm, SubmitInjectedComponent} from '@yarnpkg/libui/sources/misc/renderForm';
+import {Command, Usage}                      from 'clipanion';
+import InkTextInput, {InkTextInputProps}     from 'ink-text-input';
+import {Box, Text, Color}                    from 'ink';
+import React, {useEffect, useState}          from 'react';
+
+import {AlgoliaPackage, search}              from '../algolia';
+
+const targets = [`regular`, `dev`, `peer`];
+
+// eslint-disable-next-line arca/no-default-export
+export default class SearchCommand extends BaseCommand {
+  static usage: Usage = Command.Usage({
+    category: `Interactive commands`,
+    description: `open the search interface`,
+    details: `
+    This command opens a fullscreen terminal interface where you can search for and install packages from the npm registry.
+    `,
+    examples: [[
+      `Open the search window`,
+      `yarn search`,
+    ]],
+  });
+
+  @Command.Path(`search`)
+  async execute() {
+    const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
+
+    const prettyName = (name: string) => {
+      const hasScope = name.includes(`/`);
+
+      if (hasScope) {
+        const [scope, descriptor] = name.split(`/`);
+        return <>{configuration.format(`${scope}/`, FormatType.SCOPE)}{configuration.format(descriptor, FormatType.NAME)}</>;
+      } else {
+        return configuration.format(name, FormatType.NAME);
+      }
+    };
+
+    const Prompt = () => {
+      return (
+        <Box flexDirection="row">
+          <Box flexDirection="column" width={49}>
+            <Box marginLeft={1}>
+             Press <Color bold cyanBright>{`<up>`}</Color>/<Color bold cyanBright>{`<down>`}</Color> to move between packages.
+            </Box>
+            <Box marginLeft={1}>
+             Press <Color bold cyanBright>{`<space>`}</Color> to select a package.
+            </Box>
+            <Box marginLeft={1}>
+              Press <Color bold cyanBright>{`<space>`}</Color> again to change the target.
+            </Box>
+          </Box>
+          <Box flexDirection="column">
+            <Box marginLeft={1}>
+              Press <Color bold cyanBright>{`<enter>`}</Color> to install the selected packages.
+            </Box>
+            <Box marginLeft={1}>
+             Press <Color bold cyanBright>{`<ctrl+c>`}</Color> to abort.
+            </Box>
+          </Box>
+        </Box>
+      );
+    };
+
+    const SearchColumnNames = () => {
+      return <>
+        <Box width={17}><Color bold underline gray>Owner</Color></Box>
+        <Box width={17}><Color bold underline gray>Version</Color></Box>
+      </>;
+    };
+
+    const SelectedColumnNames = () => {
+      return <Box width={17}><Color bold underline gray>Target</Color></Box>;
+    };
+
+    const HitEntry = ({hit, active}: {hit: AlgoliaPackage, active: boolean}) => {
+      const [action, setAction] = useMinistore<string | null>(hit.name, null);
+
+      useSpace({
```

Comment:
I'm not sure if using `<space>` both for selecting packages and targets is a good idea. Wouldn't it be better to use `<left>/<right>` to navigate through targets instead? (Again, I'm just wondering, don't take it as "you should do it this way" 🙂)
Also, I haven't checked: Are you supposed to also deselect a package using space?

Reply:
Hmm, I'm not sure. I thought about using `<left>/<right>` keys, but then I feel like there should be a way for you to put the focus on the bottom part, the selected packages, and change targets from there, but honestly, that just seems more confusing because of the added complexity.

I still feel like how it is now is best, but if you want me to change it so we can all test it out, I definitely can do it.

Reply:
After using it a bit more, I think it's fine as it is now, it feels more intuitive than I originally thought. This can be marked as resolved if nobody else has a different opionion.

Reply:
Resolving it then.